src/cloud-cli/meltano/cloud/api/client.py

```diff
@@ -131,13 +132,21 @@ def headers(self, headers: dict) -> t.Iterator[None]:
     async def authenticated(self) -> t.AsyncIterator[None]:
         """Provide context for API calls which require authentication.
 
+        Authorization and ID tokens may be out-of-date. In this case, requests made
+        within the context which get a 403 response will cause re-authentication, and
+        the request will be remade.
+
+        Note: Will not work correctly if nested.
```

Comment:
The technical term for this (used widely in the Python stdlib docs) is [reentrant](https://docs.python.org/3/library/contextlib.html#reentrant-context-managers)

```suggestion
        This context manager is not reentrant.
```